cumulus/parachains/integration-tests/chopsticks/tests/v5-tests/index.ts

```diff
@@ -0,0 +1,245 @@
+	expect(bobBalanceAfter > bobBalanceBefore).toBeTruthy();
+});
+
+test("Initiate Teleport XCM v4", async () => {
+	const msg = Enum("V4", [
+		XcmV4Instruction.WithdrawAsset([{
+			id: { parents: 1, interior: XcmV3Junctions.Here() },
+			fun: XcmV3MultiassetFungibility.Fungible(7e12),
+		}]),
+		XcmV4Instruction.SetFeesMode({
+			jit_withdraw: true,
+		}),
+		XcmV4Instruction.InitiateTeleport({
+			assets: XcmV4AssetAssetFilter.Wild({type: "All", value: undefined}),
+			dest: {
+				parents: 1,
+				interior: XcmV3Junctions.Here(),
+			},
+			xcm: [
+				XcmV4Instruction.BuyExecution({
+					fees: {
+						id: { parents: 0, interior: XcmV3Junctions.Here() },
+						fun: XcmV3MultiassetFungibility.Fungible(500_000_000_000n),
+					},
+					weight_limit: XcmV3WeightLimit.Unlimited(),
+				}),
+			],
+		}),
+	]);
+	const weight = await AHApi.apis.XcmPaymentApi.query_xcm_weight(msg);
+
+	const ahToWnd = AHApi.tx.PolkadotXcm.execute({
+			message: msg,
+			max_weight: { ref_time: weight.value.ref_time, proof_size: weight.value.proof_size },
+		},
+	);
+
+	const r = await ahToWnd.signAndSubmit(aliceSigner);
+	expect(r).toBeTruthy();
+})
+
+test("Initiate Teleport XCM v5", async () => {
+	const msg = Enum('V5', [
+		XcmV4Instruction.WithdrawAsset([
+			{
+				id: { parents: 1, interior: XcmV3Junctions.Here() },
+				fun: XcmV3MultiassetFungibility.Fungible(5_000_000_000_000n),
+			},
+
+		]),
+		Enum('PayFees', {
+			asset: {
+				id: {
+					parents: 1,
+					interior: XcmV3Junctions.Here(),
+				},
+				fun: XcmV3MultiassetFungibility.Fungible(2_000_000_000_000n),
+			}
+		}),
+		Enum('InitiateTransfer', {
+			destination: {
+				parents: 1,
+				interior: XcmV3Junctions.Here(),
+			},
+			// optional field. an example of usage:
+			// remote_fees: Enum('Teleport', {
+			// 	type: 'Wild',
+			// 	value: {
+			// 		type: 'All',
+			// 		value: undefined,
+			// 	},
+			// }),
+			preserve_origin: false,
+			assets: [Enum('Teleport', {
+				type: 'Wild',
+				value: {
+					type: 'All',
+					value: undefined,
+				},
+			})],
+			remote_xcm: [
+				Enum('PayFees', {
+					asset: {
+						id: {
+							parents: 0,
+							interior: XcmV3Junctions.Here(),
+						},
+						fun: XcmV3MultiassetFungibility.Fungible(1_000_000_000_000n),
+					}
+				}),
+				XcmV4Instruction.DepositAsset({
+					assets: XcmV3MultiassetMultiAssetFilter.Definite([{
+						fun: XcmV3MultiassetFungibility.Fungible(1_000_000_000_000n),
+						id: { parents: 0, interior: XcmV3Junctions.Here() },
+					}]),
+					beneficiary: {
+						parents: 0,
+						interior: XcmV3Junctions.X1(XcmV3Junction.AccountId32({
+							network: undefined,
+							id: Binary.fromBytes(hdkdKeyPairBob.publicKey),
+						})),
+					},
+				}),
+			],
+		}),
+	]);
+	const weight = await AHApi.apis.XcmPaymentApi.query_xcm_weight(msg);
+
+	const ahToWnd = AHApi.tx.PolkadotXcm.execute({
+			message: msg,
+			max_weight: { ref_time: weight.value.ref_time, proof_size: weight.value.proof_size },
+		},
+	);
```

Comment:
Awesome use of the runtime API, I didn't think of it but it's much easier now for clients to call "execute". Did you need to add some extra weight as margin or did this work out of the box?

Reply:
This worked out of the box. I thought that was the intention 😸